### ISSUE TITLE
REFERENCE rosters not handled

### ISSUE THREAD

**Issue by paul90** [WardCunningham/search#2]
The scraper does not appear to handle ROSTER items that are using a `REFERENCE` to populate the roster.

For example see `fedwiki.rodwell.me/paul-rodwell.html` where the roster uses `REFERENCE` to load `rodwell.me/paul-rodwells-sites`.

**Comment by paul90** [comment]
Also does not follow ROSTERS like `ROSTER goals.pods.wiki.org/goals-for-federated-wiki`

**Comment by WardCunningham** [comment]
Can you think of an elegant way to exploit the logic already implemented in the Roster plugin?

**Comment by paul90** [comment]
How about changing that regex from `/^(([a-zA-Z0-9-]+(\.[a-zA-Z0-9-]+)+)(:\d+)?)$/` to `/(([a-zA-Z0-9-]+(\.[a-zA-Z0-9-]+)+)(:\d+)?)/` this would then find site names wherever they are on the Roster plugin lines.

I also seem to be missing something - trying this I find that while the site is discovered, it appears in `sites.txt`, it is not scraped.

**Comment by WardCunningham** [comment]
Brilliant.

Somehow in my mind I was thinking that I had to duplicate the indirect lookup provided by the advanced Roster markup while I was reading the item. But you correctly point out is all I have to do is notice the sites then the basic scraper logic does the lookup on its own.

I will make the suggested change.

**Comment by WardCunningham** [comment]
Yes, you are also correct to notice that the scrape does not yet expand when it notices new sites. You can add new sites by adding an appropriately named subdirectory in sites. I do this manually with a command something like:

```
cd sites; mkdir `cat ../sites.txt`
```

You can diff the sites.txt and the actual sites scraped to see what's missing.

```
diff sites.txt <(ls sites)
```

One reason for this caution is that the scraper would tend to find sites I didn't want scraped, such as private sites on my own laptop where I was developing the scraper. Even now, I run it on an old laptop on my LAN where it can see sites that aren't public. Once I learn all the checks I need I will code them into the scraper itself and let it make new subdirectories on its own.

**Comment by WardCunningham** [comment]
Fixed by c697661721881d7a569274600c386a619f59ab1b

Oops. I should have fixed this before the noon scrape. Then I could propagate the sites I've missed this afternoon. Sorry to be so slow.

**Comment by WardCunningham** [comment]
Brilliant, but not identical.

I realize there is a small difference in what I was thinking I should scrape and what I am now scraping from REFERENCE and ROSTER keywords.

Since I don't learn about the indirect sites until that page is processed later in the scrape, I don't credit the page with having cited those sites. I suppose this is ok. In a gray area I tend to side with what the computer makes natural to code.
